pkgs/by-name/ho/homebridge-config-ui-x/package.nix

```
@@ -0,0 +1,62 @@
+{
+  lib,
+  pkgs,
+  buildNpmPackage,
+  fetchFromGitHub,
+  fetchNpmDeps,
+  npmHooks,
+}:
+
+buildNpmPackage (finalAttrs: {
+  pname = "homebridge-config-ui-x";
+  version = "4.75.0";
+
+  src = fetchFromGitHub {
+    owner = "homebridge";
+    repo = "homebridge-config-ui-x";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-zmr35ejcON5haUKkXiLBqAVSkP26QYVc3Ftnc76KnYw=";
+  };
+
+  # Deps hash for the root package
+  npmDepsHash = "sha256-aqG9GQ2gdrCz+lxTOVYFo96jusb710KkguN/5R0Ye6c=";
+
+  # Need to also run npm ci in the ui subdirectory
+  preBuild =
+    let
+      # Deps src and hash for ui subdirectory
+      npmDeps_ui = fetchNpmDeps {
+        name = "npm-deps-ui";
+        src = "${finalAttrs.src}/ui";
+        hash = "sha256-WHGg2sINF1fH3tnuZsf+yjqoyWSPU5osOhuZb5VU+8I=";
+      };
+    in
+    ''
+      # Tricky way to run npmConfigHook multiple times
+      (
+        source ${npmHooks.npmConfigHook}/nix-support/setup-hook
+        npmRoot=ui npmDeps=${npmDeps_ui} makeCacheWritable= npmConfigHook
+      )
```

```suggestion
  npmDepsHash = "sha256-aqG9GQ2gdrCz+lxTOVYFo96jusb710KkguN/5R0Ye6c=";

  # Deps src and hash for ui subdirectory
  npmDeps_ui = fetchNpmDeps {
    name = "npm-deps-ui";
    src = "${finalAttrs.src}/ui";
    hash = "sha256-WHGg2sINF1fH3tnuZsf+yjqoyWSPU5osOhuZb5VU+8I=";
  };

  # Need to also run npm ci in the ui subdirectory
  preBuild =  
    ''
      # Tricky way to run npmConfigHook multiple times
      (
        source ${npmHooks.npmConfigHook}/nix-support/setup-hook
        npmRoot=ui npmDeps=${finalAttrs.npmDeps_ui} makeCacheWritable= npmConfigHook
      )
```

something like this.